backend/src/core/services/campaign.service.ts

```diff
@@ -144,25 +150,95 @@ const listCampaigns = ({
   userId,
   offset,
   limit,
+  type,
+  status,
+  name,
+  sortBy,
+  orderBy,
 }: {
   userId: number
   offset: number
   limit: number
+  type?: ChannelType
+  status?: Status
+  name?: string
+  sortBy?: SortField
+  orderBy?: Ordering
 }): Promise<{ rows: Array<Campaign>; count: number }> => {
   const campaignJobs = '(PARTITION BY "job_queue"."campaign_id")'
   const maxAge = config.get('redaction.maxAge')
 
+  let whereFilter: {
+    user_id: number
+    type?: ChannelType
+  } = { user_id: userId }
+
+  if (type) {
+    whereFilter.type = type
+  }
+
+  if (name) {
+    const nameMatch = {
+      name: {
+        [Op.iLike]: '%' + name + '%',
+      },
+    }
+    whereFilter = { ...whereFilter, ...nameMatch }
+  }
+
+  if (status) {
+    let operation: any = {}
+    switch (status) {
+      case Status.Draft: {
+        operation = { [Op.is]: null }
+        break
+      }
+      case Status.Sending: {
+        operation = {
+          [Op.in]: [
+            JobStatus.Ready,
+            JobStatus.Enqueued,
+            JobStatus.Sending,
+            JobStatus.Sent,
+            JobStatus.Stopped,
+          ],
+        }
+        break
+      }
+      case Status.Sent: {
+        operation = { [Op.eq]: JobStatus.Logged }
+        break
+      }
+      default: {
+        break
+      }
+    }
+    const statusFilter = { '$job_queue.status$': operation }
+    whereFilter = { ...whereFilter, ...statusFilter }
+  }
+
+  let orderOpt: any = [['created_at', 'DESC']]
```

Comment:
think we spoke about this earlier; wonder if we can give it a more specific type hm